script.py

```diff
@@ -0,0 +1,239 @@
+"""To use this script, install the provided requirements.txt file and
+supply a secrets.py file with the required parameters listed below.
+
+JSON records will be written into the specified file and HTML reports
+will be written into an outputs filter under the provided directory.
+"""
+
+
+from datetime import datetime, timedelta
+from jinja2 import Template
+from ua_parser import user_agent_parser
+
+import collections
+import json
+import os
+import requests
+import secrets
+import smtplib
+
+
+# This is the absolute path to the directory where you'd like
+# the JSON and HTML files to be written.
+# The template HTML file should also be placed in this
+# directory.
+DIR = secrets.DIR
```

Comment:
Does it behave reasonably if these configs aren't supplied? It should definitely abort with a message if missing user/pass/from/to email, but for the directory/filename could we pick pick sensible defaults.

Reply:
The directory needs to be an absolute path if this is run via cron so we would need something like `/Users/...` I could just use the directory in which the script lives, but I'm not sure using at as a default is good. What do you think?

Reply:
I don't see anything particularly wrong with the script's directory as a default. The problem with anything outside of the script directory is that we can't count on the presence of any paths e.g. `/Users/` on other platforms. Even if it was `~/something` that would break on Windows, right?

So the choice would be between refusing to run without a supplied output directory, or using `__file__/output`. I'd say the latter seems like the better choice, unless I'm missing something.

Reply:
Agreed on that we can't count on a path existing. I guess the latter is fine, although I might hesitate to write files somewhere without explicit choice. Seems reasonable in this case though, so I'll implement that.